code/ui/alivehud/inspectmenu/InspectMenu.cs

```diff
@@ -132,6 +151,88 @@ public void SetPlayer(TTTPlayer player)
                     _playerImage.Style.BorderColor = player.Role.Color;
                     _playerImage.Style.Dirty();
                 }
+
+                public void SetConfirmationData(ConfirmationData confirmationData)
+                {
+                    _confirmationData = confirmationData;
+
+                    _timeSinceDeath?.Delete(true);
+
+                    _timeSinceDeath = new InspectItem(this);
+                    _timeSinceDeath.ImageWrapper.Image.SetTexture($"/ui/inspectmenu/time.png");
+                    _timeSinceDeath.InspectItemLabel.Text = "";
+
+                    _headshot?.Delete(true);
+
+                    if (confirmationData.Headshot)
+                    {
+                        _headshot = new InspectItem(this);
+                        _headshot.ImageWrapper.Image.SetTexture($"/ui/inspectmenu/headshot.png");
+                        _headshot.InspectItemLabel.Text = "By a headshot";
+                    }
+
+                    _distance?.Delete(true);
+                    _suicide?.Delete(true);
+
+                    if (!confirmationData.Suicide)
+                    {
+                        _distance = new InspectItem(this);
+                        _distance.ImageWrapper.Image.SetTexture($"/ui/inspectmenu/distance.png");
+                        _distance.InspectItemLabel.Text = $"From {confirmationData.Distance:n0}m away";
+                    }
+                    else
+                    {
+                        _suicide = new InspectItem(this);
+                        _suicide.ImageWrapper.Image.SetTexture("");
+                        _suicide.InspectItemLabel.Text = $"Committed suicide";
+                    }
```

Comment:
Can we not just update the data instead of deleting and re-creating these panels?

Reply:
We could but things not always exist for every confirmation stuff and then it could get reordered. That here makes sure that it's always in the normal order.

Reply:
I'm not sure I understand, if the components are created once, and then updated, how could they end up being re-ordered?

Reply:
As there gonna be a lot of ui icon elements, hiding is not an option. So we have to delete them. And if you e.g. delete the headshot entry, it will - if there is an headshot on another corpse - be created and *appended* to the end of the element (instead of being placed in the mid / as second or third element e.g.).